applications/ShallowWaterApplication/custom_conditions/conservative_condition.h

```diff
@@ -0,0 +1,244 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_CONSERVATIVE_CONDITION_H_INCLUDED
+#define KRATOS_CONSERVATIVE_CONDITION_H_INCLUDED
+
+// System includes
+
+
+// External includes
+
+
+// Project includes
+#include "wave_condition.h"
+
+namespace Kratos
+{
+///@addtogroup ShallowWaterApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @ingroup ShallowWaterApplication
+ * @class ConservativeCondition
+ * @brief Implementation of a condition for shallow water waves problems
+ * @author Miguel Maso Sotomayor
+ */
+template<std::size_t TNumNodes>
+class ConservativeCondition : public WaveCondition<TNumNodes>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    typedef std::size_t IndexType;
+
+    typedef Node<3> NodeType;
+
+    typedef Geometry<NodeType> GeometryType;
+
+    typedef WaveCondition<TNumNodes> WaveConditionType;
+
+    typedef typename WaveConditionType::NodesArrayType NodesArrayType;
+
+    typedef typename WaveConditionType::PropertiesType PropertiesType;
+
+    typedef typename WaveConditionType::EquationIdVectorType EquationIdVectorType;
+
+    typedef typename WaveConditionType::DofsVectorType DofsVectorType;
+
+    typedef typename WaveConditionType::ConditionData ConditionData;
+
+    typedef typename WaveConditionType::LocalVectorType LocalVectorType;
+
+    typedef typename WaveConditionType::LocalMatrixType LocalMatrixType;
+
+    ///@}
+    ///@name Pointer definition
+    ///@{
+
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(ConservativeCondition);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /**
+     * @brief Default constructor
+     */
+    ConservativeCondition() : WaveConditionType(){}
+
+    /**
+     * @brief Constructor using an array of nodes
+     */
+    ConservativeCondition(IndexType NewId, const NodesArrayType& ThisNodes) : WaveConditionType(NewId, ThisNodes){}
+
+    /**
+     * @brief Constructor using Geometry
+     */
+    ConservativeCondition(IndexType NewId, GeometryType::Pointer pGeometry) : WaveConditionType(NewId, pGeometry){}
+
+    /**
+     * @brief Constructor using Geometry and Properties
+     */
+    ConservativeCondition(IndexType NewId, GeometryType::Pointer pGeometry, typename PropertiesType::Pointer pProperties) : WaveConditionType(NewId, pGeometry, pProperties){}
+
+    /**
+     * @brief Destructor
+     */
+    virtual ~ ConservativeCondition(){}
```

Comment:
Minor comment, but if this is a condition should override

Reply:
thanks, I'm going to modify it.